test/test.hs

```diff
@@ -2448,6 +2417,82 @@ instance Arbitrary (LitOnly (Expr EWord)) where
 instance Arbitrary (LitOnly (Expr Buf)) where
   arbitrary = LitOnly . ConcreteBuf <$> arbitrary
 
+-- GenWordSimplifyExpr
+newtype GenWordSimplifyExpr a = GenWordSimplifyExpr a
+  deriving (Show, Eq)
+
+instance Arbitrary (GenWordSimplifyExpr (Expr EWord)) where
+  arbitrary = do
+    fmap GenWordSimplifyExpr . sized $ genWord 0
+
+-- GenWriteWordBuf
+newtype GenWriteWordBuf a = GenWriteWordBuf a
+  deriving (Show, Eq)
+
+instance Arbitrary (GenWriteWordBuf (Expr Buf)) where
+  arbitrary = do
+    let mkBuf = oneof
+          [ pure $ ConcreteBuf ""       -- empty
+          , fmap ConcreteBuf arbitrary  -- concrete
+          , sized (genBuf 100)          -- overlapping writes
+          , arbitrary                   -- sparse writes
+          ]
+    fmap GenWriteWordBuf mkBuf
+
+-- GenCopySliceBuf
+newtype GenCopySliceBuf a = GenCopySliceBuf a
+  deriving (Show, Eq)
+
+instance Arbitrary (GenCopySliceBuf (Expr Buf)) where
```

Comment:
Same question here about making this the default?

Reply:
We decided it's not so important so I'm resolving this.